### ISSUE TITLE
Is this extension still maintained?

### ISSUE THREAD

**Issue by PikaDude** [animal-crossing-music-extension/ac-music-extension#47]
Hello! I recently came up with an idea to make some sort of program that would play the associated Animal Crossing music tracks in real-time. Turns out that already exists (right here), but I've noticed that development and maintaining has slowed or almost stopped, and that the extension on the Chrome Store hasn't been updated in a few years. I also noticed that this extension was in potential need of developers or something along those lines. I've made a Chrome Extension and also a Firefox extension in the past and have pretty good experience with JS, HTML and CSS (my profile is a bad example, I don't have a lot of open-source projects) and I'd be down to lend a hand with this whenever possible.

**Comment by PikaDude** [comment]
After cloning the repo I can see why the extension hasn't been updated on the Chrome Webstore lol. Although, after a bit of tinkering I seem to have gotten most of it working again and I'd be more than happy to fix things up around here to have a stable version.

**Comment by JdotCarver** [comment]
Hey there,
Yeah I never had the skills to figure out all of this by myself and other projects came by and swept this one under the rug. I'd be happy to have you help out! A lot of people have contributed to this project and their work remains unpublished because of the conflicts I'm unable to resolve. But if you say you managed, that would be great.

**Comment by PikaDude** [comment]
Is it maybe possible for me to receive write access so I can merge pull requests and maybe make commits directly? It'd make things a lot easier.

**Comment by jordigarcl** [comment]
I would also **love** to be able to contribute to this project! Found out about it on Twitter and have been lurking the repo for a couple days. 

I noticed that there is already a perfectly working version published on the Chrome Store, but the new contributions have not been published and the current state of the project crashes.

It is a pity that the work that has been contributed to this project remains unpublished due to (understandable) lack of experience and manpower! I think that, with some proper maintenance, we should be able to fix the issues with the current state of `master` so it can be properly published to the store. After all, most of the bugs that make the project crash in its current state seem to be trivial.

If the current project owner is not able to actively maintain the project anymore, perhaps we should evaluate an ownership change. Further work on the extension could continue as a new, separately published extension (honouring the original project license) and the owner could point new users/contributors to the new extension/repository, while maintaining the integrity of this project.

**Comment by PikaDude** [comment]
I agree with the ownership change, personally. However, it is not needed to create a new separate published extension as the original license only says that if we are to modify it then we must have it under a different name (perhaps append a word to the title like "revived") and make it clear that it is a modified version by doing so. We could just change the name of the extension and be done with that.

**Comment by jordigarcl** [comment]
I agree that updating the currently published extension would be the best option to continue development, as we would be able to deliver all unpublished features to the existing user-base. 

However, given that the latest Chrome version is almost 4 years old, I'm willing to bet that the maintainers are reluctant to update an already stable version, specially if they feel that they can't ensure its integrity. 

In addition to fixing the current issues that break the extension, I personally want to rework the project: updating the extension, reworking the repo readme, cleaning up opened issues, updating the chrome extension page, refactoring parts of the code and even creating a simple GitHub Page for end-users.

Since these kind of changes need a higher degree of control and involvement than the one than the one possible through PRs, I propose the following:

- We could fork the repository and work on all things mentioned above. 

- Once we reach a stable version, we can evaluate either updating the original repo with all changes from the fork, or making this new repo the de-facto one. 

- Ideally we could publish all new changes to the existing extension, and if you guys are unwilling to do that, then we could publish a new one and at least markthe original as "discontinued",  mentioning the new one in its description.

TL;DR: We want to take an active role in the project maintainance/revival. Collaboration of the current project owners would enable us to deliver these new  changes to the existing userbase. I have proposed a few ways to do this safely! 😉

**Comment by PikaDude** [comment]
Sounds good. I've gone ahead and [forked the repository here](https://github.com/PikaDude/Animal-Crossing-Music-Extension) and I've also made some small refinements, fixes and changes to get it working again. I'm currently in the process of re-organizing all of the music in their folders, getting event music working properly again, and updating the town tune sounds to be more accurate to the game. I have access to a hacked 3DS so I'm using that to dump any needs sounds from my copies of the games. I also have a lot of experience with bulk editing sounds so I'm also adjusting the format of and increasing the quality of the songs according to issue #9.

**Comment by JdotCarver** [comment]
Well I can say it's pleasant to see you working things out! Tell me if I can do anything to help you in the process. 👍

**Comment by anokhee** [comment]
Hey @PikaDude, I'd love to help contribute to this project. I can try to update the UI to make it look similar to the in-game Town Tune editor.

**Comment by PikaDude** [comment]
Sure, @anokhee. You're more than welcome to contribute to it. Just fork my fork and make a PR whenever you're ready.

**Comment by paladique** [comment]
Hi everyone! I wanted to contribute over the summer but work got super busy, but @PikaDude I also have write access and can add ya!

**Comment by paladique** [comment]
@PikaDude I went ahead and added you with write access, I like your ideas of reviving this. You should have enough permissions to triage issues and PRs and push access.

**Comment by PikaDude** [comment]
Thanks, @paladique, I'll probably continue in my own repo for now and merge the changes once they're all finished.

**Comment by paladique** [comment]
Sounds good. I believe I have access to publish as well, so we can work through that once it's time and you think it's in a good spot. :) I'll check out your repo and help out too when I can.

**Comment by paladique** [comment]
@JordiGarcL I see you're working on this too, so I'm giving ya write access!

**Comment by paladique** [comment]
Finally, the last thing I'll do is update the readme with a temporary note that development is currently happening in that repo, in case more collaborators are interested!

**Comment by jordigarcl** [comment]
@paladique This is great! I agree the best course of action is to keep working on @PikaDude 's fork and then integrate everything here once it is ready to go.

In the meanwhile we can trim this repo issues as you have already done 👍

**Comment by paladique** [comment]
@JdotCarver 👋! so we are almost ready to publish a new version, however none of us have access to to Google group to do this. You sent me an invitation over the summer but apparently I never accepted and it expired 😭  Could you provide us access? I'll still use the same email as last time: jasminegweb@gmail.com. @JordiGarcL and @PikaDude you can share yours here too or I'll invite you when I accept mine.

**Comment by jordigarcl** [comment]
Mine is jgarcialisson@gmail.com 👍
@JdotCarver you may have a look at PikaDude's repo to see the improvements made.
 @PikaDude deserves most of the credit, huge kudos to him!

**Comment by paladique** [comment]
Oh! and @JdotCarver is there a place where we can communicate with you from time to time with questions if we have em? email? [the discord group](https://discordapp.com/invite/4FMrEF8)?

**Comment by PikaDude** [comment]
My email is mail@pikadude.me
Also thanks, @JordiGarcL! I honestly appreciate the help everyone's given too.

**Comment by JdotCarver** [comment]
@paladique Just joined the Discord Group. I'm connected 80% of the day so you'll be sure to find me there. :)

**Comment by PikaDude** [comment]
Hey, @JdotCarver, any updates on giving us permission to publish the extension? We are just about finished.

**Comment by JdotCarver** [comment]
Invitations to the Google Publishing Group sent! 
![chrome_85LSmKh5Iw](https://user-images.githubusercontent.com/7091684/73047506-4e5cf780-3e76-11ea-9a42-31fc54ec13ce.png)

And @paladique you are already in the Google Group as indicated in the members page here:

![chrome_iGpPEctX2v](https://user-images.githubusercontent.com/7091684/73047524-616fc780-3e76-11ea-8aeb-55866c46e9cd.png)

PS: I also moved your status to Owner of the group.

**Comment by PikaDude** [comment]
Thank you for the invite!

**Comment by jordigarcl** [comment]
How do we stand in this issue? Now that we have merged a lot of changes into the repo, I'd say the extension is actively maintained again :)

**Comment by PikaDude** [comment]
Yeah, we should be good to close this now.
